azure/services/aso/aso.go

```diff
@@ -150,6 +158,10 @@ func (r *reconciler[T]) CreateOrUpdateResource(ctx context.Context, spec azure.A
 	parameters.SetName(resourceName)
 	parameters.SetNamespace(resourceNamespace)
 
+	if err := controllerutil.SetControllerReference(r.owner, parameters, r.Client.Scheme()); err != nil {
+		return zero, errors.Wrap(err, "failed to set owner ref")
+	}
```

Comment:
The reason we're not adding the owner ref only on create like were adding the label before is because we want to set the owner ref also on existing resources during a CAPZ upgrade.